master/internal/command/command.go

```diff
@@ -311,6 +311,10 @@ func (c *command) receiveSchedulerMsg(ctx *actor.Context) error {
 // 1. Command is aborted before being allocated.
 // 2. Forcible terminating a command by killing containers.
 func (c *command) terminate(ctx *actor.Context) {
+	if !c.task.CanTerminate {
```

Comment:
Yeah I thought about removing this entirely, but I thought some notion of preemptible commands might come up in the future and wanted to leave the correct handling.

But there's a bigger issue: in the e2e tests that ran last night, the 2 web failures were failures to kill these commands manually. I seem to be able to to kill stuff manually with my patch from the web UI, but it's not clear to me why the e2e tests would be going through a different code path than my manual testing: both appear to submit a DELETE request to the same API endpoint. I need to look into this. We can definitely make the scheduler not preempt something, but we should ensure you can still manually kill tasks too.

Reply:
> Yeah I thought about removing this entirely, but I thought some notion of preemptible commands might come up in the future and wanted to leave the correct handling.

If that comes up in the future, we just need to handle that differently in receiving `scheduler.ReleaseResources`. The reason I don't like using `c.task.CanTerminate` is that we create a dependency from `terminate` to `c.task`, which is only used for keeping track of the task state inside the scheduler. It might be fine for now since `c.task` won't be nil through the lifecycle of command but it is a bit weird to write it this way. What do you think?

> But there's a bigger issue: in the e2e tests that ran last night, the 2 web failures were failures to kill these commands manually. I seem to be able to to kill stuff manually with my patch from the web UI, but it's not clear to me why the e2e tests would be going through a different code path than my manual testing: both appear to submit a DELETE request to the same API endpoint. I need to look into this. We can definitely make the scheduler not preempt something, but we should ensure you can still manually kill tasks too.

I did see there are two code paths to kill a command. One is through `handleAPIRequest` and the other one is through `*apiv1.KillNotebookRequest`, `*apiv1.KillCommandRequest`, and messages like that. The second one is introduced in a new API that is based on protobuf and the first one, which e2e tests might depend on, has been used in the past.

Reply:
Ah ha - thank you for the pointers to the other entry points. That makes sense now - so I agree, leaving terminate() as is and just removing the ReleaseResources handling in the Receive() switch statement is best. That is reflect in my latest patch, that I need to rebase and resolve conflicts on...